### PR TITLE
[LW-19] Some fixes regarding pure logging

### PR DESCRIPTION
**Pull request by Martoon-00**
Initial purpose of this issue was to remove some boilerplate, but now I realize that I don't see a nice way to do that. After some discussion with @ChShersh we decided that it would be better to remain things as they are.
As result of this issue I removed redundant constraint in `dispatchEvents`, that's it :/

### CODE REVIEW

File: src/Pos/Ssc/GodTossing/Shares.hs [hunk 1]
```diff
@@ -47,8 +48,7 @@ getOurShares ourKey = do
                     logWarning $ sformat ("Failed to deserialize share for " % build) id
                     return Nothing
         resHM = HM.fromList . catMaybes $ res
-    loggerName <- getLoggerName
-    liftIO $ usingLoggerName loggerName $ dispatchEvents pLog
```
Comment (gromakovsky) [inline src/Pos/Ssc/GodTossing/Shares.hs]: Do I understand correctly that `usingLoggerName` was completely redundant here?

Comment (Martoon-00) [inline src/Pos/Ssc/GodTossing/Shares.hs]: Yep, it was used to satisfy mentioned redundant constraint

File: src/Pos/Ssc/Extra/Logic.hs [hunk 2]
```diff
@@ -73,18 +76,14 @@ sscRunLocalQuery action = do
 sscRunLocalSTM
     :: forall ssc m a.
        (MonadSscMem ssc m, MonadIO m, WithLogger m)
-    => (LoggerNameBox (PureLogger (StateT (SscLocalData ssc) STM)) a) -> m a
+    => NamedPureLogger (StateT (SscLocalData ssc) STM) a -> m a
 sscRunLocalSTM action = do
-    loggerName <- getLoggerName
     localVar <- sscLocal <$> askSscMem
```
Comment (Martoon-00) [inline src/Pos/Ssc/Extra/Logic.hs]: Does this change seems reasonable?

Comment (gromakovsky) [inline src/Pos/Ssc/Extra/Logic.hs]: Ok, but add a comment why this `error` can't be touched please. I took some time for me to understand what's going on.

Comment (gromakovsky) [inline src/Pos/Ssc/Extra/Logic.hs]: (I mean why we know for sure that it won't be touched)
(I understand it, but it took some time)

File: src/Pos/Ssc/Extra/Logic.hs [hunk 3]
```diff
@@ -73,18 +76,14 @@ sscRunLocalQuery action = do
...
-            oldLD <- readTVar localVar
-            ((res, events), !newLD) <-
-                flip runStateT oldLD . runPureLog . usingLoggerName loggerName $
-                action
-            (res, events) <$ writeTVar localVar newLD
-    res <$ dispatchEvents events
+    launchNamedPureLog (atomically . (`evalStateT` error "Don't touch")) $ do
+        put =<< (lift $ lift $ readTVar localVar)
+        res <- action
+        get >>= (lift . lift . writeTVar localVar)
```
Comment (volhovM) [inline src/Pos/Ssc/Extra/Logic.hs]: http://giphy.com/gifs/dancing-halloween-pumpkin-dance-l3vR4IfRCoz8Qz9oA
S Y M M E T R Y

File: stack.yaml [hunk 4]
```diff
@@ -54,6 +54,10 @@ packages:
     git: https://github.com/input-output-hk/cardano-crypto
     commit: 838b064d8a59286142aa2fe14434fe7601896ddb
   extra-dep: true
+- location:
```
Comment (volhovM) [inline stack.yaml]: Why not release on hackage?

**Comment by Martoon-00** [comment]
Okay, it's ready to be merged

File: src/Pos/Ssc/Extra/Logic.hs [hunk 5]
```diff
@@ -181,34 +183,16 @@ type SscGlobalVerifyMode ssc m =
...
-    res <$ dispatchEvents events
+    launchNamedPureLog (atomically . evalStateTUninitialized) $
+        syncingStateWith globalVar $ switchMonadBaseIdentityToSTM action
   where
-    sscRunGlobalUpdateDo loggerName globalVar = do
-        oldState <- readTVar globalVar
-        let (res, !newState, events) =
-                sscRunGlobalUpdatePure @ssc loggerName oldState action
-        (res, events) <$ writeTVar globalVar newState
+    switchMonadBaseIdentityToSTM = hoist $ hoist $ return . runIdentity
```
Comment (ChShersh) [inline src/Pos/Ssc/Extra/Logic.hs]: `return . runIdentity` is [`generalize`](http://hackage.haskell.org/package/mmorph-1.0.9/docs/Control-Monad-Morph.html#v:generalize) from `mmorph`

File: src/Pos/Ssc/Extra/Logic.hs [hunk 6]
```diff
@@ -181,34 +183,16 @@ type SscGlobalVerifyMode ssc m =
...
     :: forall ssc m a.
        SscGlobalApplyMode ssc m
-    => LoggerNameBox (PureLogger (State (SscGlobalState ssc))) a -> m a
+    => NamedPureLogger (State (SscGlobalState ssc)) a -> m a
 sscRunGlobalUpdate action = do
-    loggerName <- getLoggerName
     globalVar <- sscGlobal <$> askSscMem
-    (res, events) <- atomically $ sscRunGlobalUpdateDo loggerName globalVar
-    res <$ dispatchEvents events
+    launchNamedPureLog (atomically . evalStateTUninitialized) $
```
Comment (ChShersh) [inline src/Pos/Ssc/Extra/Logic.hs]: I don't quite understand how it works... `launchNamedPureLog` is a functions, that takes named logging monadic actions and returns another action. How do you able to pass two monadic arguments to it?

**Comment by Martoon-00** [comment]
Previously you said it's optional
https://serokell.slack.com/archives/log-warper/p1489572191719322
But okay, I will

**Comment by gromakovsky** [comment]
Well, I expected it to be more complicated than a simple `git revert`. I think it doesn't make sense to use `git revert` in feature branch (assuming you are the only one who is working there), because you can simply remove that commit and force push as long as there are no merge commits in between.

**Comment by Martoon-00** [comment]
I didn't want comments here to vanish )  But it's not worth it I guess

**Comment by gromakovsky** [comment]
AFAIK, comments don't vanish. I don't fully understand how all this pull requests machinery works though.

**Comment by Martoon-00** [comment]
Really, still here they are

**Pull request merged**